File: app/src/main/java/com/example/danila/minerandroid/DBHelper.java
package com.example.danila.minerandroid;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.support.annotation.Nullable;

public class DBHelper extends SQLiteOpenHelper {

    private static final int DATABASE_VERSION = 1;
    private static final String DATABASE_NAME = "Miner_database";
    static final String TABLE_NAME = "Miner_records";
    private static final String KEY_ID = "_id";
    static final String KEY_NAME = "name";
    static final String KEY_TIME = "time";

    DBHelper(@Nullable Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        String sql = "create table " + TABLE_NAME + "(" +
                KEY_ID + " integer primary key autoincrement," +
                KEY_NAME + " text not null," +
                KEY_TIME + " text not null" + ")";
        db.execSQL(sql);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        String sql = "drop table if exists " + TABLE_NAME;
        db.execSQL(sql);
        onCreate(db);
    }

}

File: app/src/main/java/lestwald/insectkiller/DBHelper.kt
package lestwald.insectkiller

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DBHelper(context: Context, private val dbName: String) : SQLiteOpenHelper(context, dbName, null, DB_VERSION) {

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("create table $dbName($KEY_ID integer primary key,$KEY_NAME text,$KEY_SCORE integer)")
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("drop table if exists $dbName")
        onCreate(db)
    }

    companion object {
        const val DB_VERSION = 1
        const val KEY_ID = "_id"
        const val KEY_NAME = "name"
        const val KEY_SCORE = "score"
    }
}

